data-model/src/main/java/io/micronaut/data/model/naming/NamingStrategy.java

```diff
@@ -156,8 +156,10 @@ default String mappedAssociatedName(@NonNull String associatedName) {
             }
         }
         if (foreignAssociation != null) {
```

Comment:
So, in some cases (when used QueryModel.from...) `==` comparison didn't work and it works ok when using `equals` which was needed to be added for `RuntimePersistentProperty`